operator/builtin/input/file/reader.go

```diff
@@ -116,7 +120,7 @@ func (r *Reader) ReadToEnd(ctx context.Context) {
 		return
 	}
 
-	scanner := NewPositionalScanner(r, r.fileInput.MaxLogSize, r.Offset, r.fileInput.SplitFunc)
+	scanner := NewPositionalScanner(r, r.fileInput.MaxLogSize, r.Offset, r.multiline.SplitFunc)
```

Comment:
I believe this is the only place we are using the split func, so I'm not sure why it needs to be part of the multiline config.

We have two concepts here:
1. How to split lines
2. When to force a flush

I'm not convinced these are dependent on each other, or that either is necessarily a concern of multiline behavior. (Of course a split func may implement multiline splitting, but it does not always.)

Can we leave splitFunc as it was and just add a new config that controls flushing?

Reply:
There is some relations I was unsure how to resolve, so I put both functionality into one struct:

 - splitFunc needs to be aware when to force a flush (so new structure is created and pass to the splitFunc)
 - every file should be forced separately (so splitFunc is created for every unique reader)
 
 Regarding line splitting. Of course it is not always a multiline behavior, but looking at it from user perspective it is intuitive to use `multline` config for it, so `Multiline` struct underneath, than mixing `Splitter` struct with `multiline` config, or using `splitter` config.
 
 I will exclude force a flush to separate config, as logic of it is mostly out of the splitFunc, but not sure how to resolve other dependencies in nice way

Reply:
I made some changes in requested direction, but after all I think `Splitter` struct seems to be reasonable consensus. 

You can see current implementation without `Splitter`. There are two dependent entities: `splitFunc` and `ForceFlush` and I feel that they should be connected together via additional struct. WDYT?

Reply:
I think consolidating into one struct makes sense

Reply:
See the latest code, please